src/client/mir_screencast.cpp

```diff
@@ -1,291 +0,0 @@
-        protobuf_screencast->has_buffer_stream() && protobuf_screencast->buffer_stream().has_buffer())
-    {
-        std::lock_guard<decltype(mutex)> lock(mutex);
-        try
-        {
-            buffer_stream = connection->make_consumer_stream(protobuf_screencast->buffer_stream());
-        }
-        catch (...)
-        {
-            callback(nullptr, context);
-            create_screencast_wait_handle.result_received();
-            return;
-        }
-    }
-
-    callback(this, context);
-    create_screencast_wait_handle.result_received();
-}
-
-void MirScreencast::released(
-    MirScreencastCallback callback, void* context)
-{
-    callback(this, context);
-
-    {
-        std::lock_guard<decltype(mutex)> lock(mutex);
-        if (buffer_stream)
-            connection->release_consumer_stream(buffer_stream.get());
-        buffer_stream.reset();
-    }
-    release_wait_handle.result_received();
-}
-
-MirBufferStream* MirScreencast::get_buffer_stream()
-{
-    std::lock_guard<decltype(mutex)> lock(mutex);
-    return buffer_stream.get();
-}
-
-void MirScreencast::screencast_done(ScreencastRequest* request)
-{
-    auto const status = request->response.has_error() ? mir_screencast_error_failure : mir_screencast_success;
-
-    if (status == mir_screencast_error_failure)
-        protobuf_screencast->set_error(request->response.error());
-
-    /*
-     * We need the request info to outlast the available_callback,
-     * but after the available_callback has been run the caller is free to call
-     * mir_screencast_release() and destroy this state.
-     *
-     * Do a two-phase remove; remove the request from the Screencast state,
-     * then call available_callback, then drop the no-longer referenced request.
-     */
-    std::unique_ptr<ScreencastRequest> request_holder;
-    {
-        std::unique_lock<decltype(mutex)> lk(mutex);
-        auto it = std::find_if(requests.begin(), requests.end(),
-                               [&request](auto const& it)
-                               { return it.get() == request; });
-        if (it != requests.end())
-        {
-            request_holder = std::move(*it);
-            requests.erase(it);
-        }
-    }
-
-    request->available_callback(status, reinterpret_cast<MirBuffer*>(request->buffer), request->available_context);
-}
-
-void MirScreencast::screencast_to_buffer(
-    mcl::MirBuffer* buffer,
-    MirScreencastBufferCallback cb,
-    void* context)
-{
-    if (!server) //construction with nullptr should be invalid
-        BOOST_THROW_EXCEPTION(std::runtime_error("invalid screencast"));
-    
-    std::unique_lock<decltype(mutex)> lk(mutex);
-
-    requests.emplace_back(std::make_unique<ScreencastRequest>(buffer, cb, context));
-
-    mir::protobuf::ScreencastRequest request;
-    request.set_buffer_id(buffer->rpc_id());
-    request.mutable_id()->set_value(protobuf_screencast->screencast_id().value());
-
-    server->screencast_to_buffer(
```

Comment:
This is fine to drop because there isn't any actual screencast logic going on. Poking around did lead me to discover that the bulk of the implementation was dropped in 35ff6b619d8, which may be useful to look at when I start trying to implement it for Wayland again.